src/runtime/HalideRuntimeCuda.h

```diff
@@ -40,18 +40,17 @@ extern int halide_cuda_run(void *user_context,
  * must be NULL when this routine is called. This call can fail due to
  * being passed an invalid device pointer. The device and host dirty
  * bits are left unmodified. */
-extern int halide_cuda_wrap_device_ptr(void *user_context, struct halide_buffer_t *buf, uintptr_t device_ptr);
+extern int halide_cuda_wrap_device_ptr(void *user_context, struct halide_buffer_t *buf, uint64_t device_ptr);
 
 /** Disconnect this halide_buffer_t from the device pointer it was
```

Comment:
I'm not sure how valid these comments are as the only memory that can be allocated is the device interface. I guess the device specific code can allocate a unique device interface per handle, but perhaps the comments should be changed a bit to reflect that these calls are mostly type conversions and device_interface allocators now. I think they are highly valuable for code documentation and searching purposes, but since we changed the dev field to be the plain API specific handle directly, we shouldn't imply the interface is more opaque than it really is.